### PR TITLE
Fix KeyError on non-existent plugin dependency

### PR DESCRIPTION
**Pull request by danlamanna**
See #980 for details of the error.

This addresses the issue by returning a 500 on the ```/system/plugins``` API call if a plugin can't be enabled. It addresses the UI issue by disabling the switch with a message if it can't enable all of the dependencies as well.

In the future, the PluginsView may need to be changed a bit - because it currently assumes that PUT calls to ```/system/plugins``` will succeed, leaving switches in an inconsistent state, restart buttons around when it doesn't need to be restarted, etc.

### CODE REVIEW

**Comment by zachmullen** [comment]
Let's add a test for this.

**Comment by danlamanna** [comment]
Should we change the possible response statuses of ```/system/setting```? It would make things more complicated, but with this change we begin to entertain the notion that changing a setting can fail.

The JS tests will fail until I have ```has_nonexistent_deps``` enabled as a plugin.

**Comment by zachmullen** [comment]
> Should we change the possible response statuses of /system/setting? It would make things more complicated, but with this change we begin to entertain the notion that changing a setting can fail.

Do you mean `PUT /system/plugins`?

**Comment by danlamanna** [comment]
```PUT /system/plugins``` is already taken care of in ebe3f17. Though ```PUT /system/setting``` could trigger the required dependencies exception if trying to set the ```core_plugins_enabled``` setting to something invalid.

**Comment by zachmullen** [comment]
Oh, did you just mean in the description? Yes, we should definitely add that it can return a 400 if the value passed in is invalid (i.e. ValidationException is raised).

**Comment by danlamanna** [comment]
@zachmullen This is ready for review.

UI Screenshot:
girder-1011-plugin-pane


The only thing I couldn't figure out was how to make ```PLUGIN_DIRS``` a proper CMake multival argument while still passing all arguments to the environment variable. I made it a colon separated string for now.

Also, do you think it should be called ```PLUGIN_DIRECTORY``` to match the option we recently added to the configuration file?

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +44,7 @@ add_web_client_test(data_filesystem "${PROJECT_SOURCE_DIR}/clients/web/test/spec
 add_web_client_test(data_gridfs "${PROJECT_SOURCE_DIR}/clients/web/test/spec/dataSpec.js" ASSETSTORE gridfs)
 add_web_client_test(data_gridfsrs "${PROJECT_SOURCE_DIR}/clients/web/test/spec/dataSpec.js" ASSETSTORE gridfsrs RESOURCE_LOCKS replicaset)
 add_web_client_test(data_s3 "${PROJECT_SOURCE_DIR}/clients/web/test/spec/dataSpec.js" ASSETSTORE s3 WEBSECURITY false)
-add_web_client_test(admin "${PROJECT_SOURCE_DIR}/clients/web/test/spec/adminSpec.js" ASSETSTORE s3 WEBSECURITY false)
+add_web_client_test(admin "${PROJECT_SOURCE_DIR}/clients/web/test/spec/adminSpec.js" ASSETSTORE s3 WEBSECURITY false PLUGIN_DIRS "${PROJECT_SOURCE_DIR}/plugins:${PROJECT_SOURCE_DIR}/tests/test_plugins")
```
Comment (zachmullen) [inline tests/CMakeLists.txt]: Let's break this line up since it's gotten so long

File: girder/api/v1/system.py [hunk 2]
```diff
@@ -149,8 +150,13 @@ def getPlugins(self, params):
         Return the plugin information for the system. This includes a list of
         all of the currently enabled plugins, as well as
         """
+        allPlugins = plugin_utilities.findAllPlugins()
+        for plugin in allPlugins.values():
+            if 'dependencies' in plugin:
+                plugin['dependencies'] = list(plugin['dependencies'])
```
Comment (zachmullen) [inline girder/api/v1/system.py]: @danlamanna do you mind if I add a commit on this branch to make it so the conversion of sets to lists is done automatically in our REST layer? Then I will remove this change.

Comment (danlamanna) [inline girder/api/v1/system.py]: Not at all.

**Comment by zachmullen** [comment]
@danlamanna I added my suggested change 0351656. Your changes LGTM except for that one too-long line. :+1:

**Comment by zachmullen** [comment]
@danlamanna Fixed my test, feel free to merge if you approve my changes.

**Pull request merged**